velox/functions/prestosql/types/IPPrefixType.cpp

```diff
@@ -51,8 +67,90 @@ class IPPrefixCastOperator : public exec::CastOperator {
       const TypePtr& resultType,
       VectorPtr& result) const override {
     context.ensureWritable(rows, resultType, result);
-    VELOX_NYI(
-        "Cast from IPPrefix to {} not yet supported", resultType->toString());
+    switch (resultType->kind()) {
+      case TypeKind::VARCHAR:
+        return castToString(input, context, rows, *result);
+      default:
+        VELOX_NYI(
+            "Cast from IPPrefix to {} not yet supported",
+            resultType->toString());
+    }
+  }
+
+ private:
+  static void castToString(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* flatResult = result.as<FlatVector<StringView>>();
+    auto rowVector = input.as<RowVector>();
+    auto rowType = rowVector->type();
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      const auto* ipaddr = rowVector->childAt(ipaddress::kIpRowIndex)
+                               ->as<SimpleVector<int128_t>>();
+      const auto* prefix = rowVector->childAt(ipaddress::kIpPrefixRowIndex)
+                               ->as<SimpleVector<int8_t>>();
+
+      const auto ipAddrVal = ipaddr->valueAt(row);
+      // The string representation of the last byte needs
+      // to be unsigned
+      const uint8_t prefixVal = prefix->valueAt(row);
+
+      // Copy the first 16 bytes into a ByteArray16.
+      folly::ByteArray16 addrBytes;
+      memcpy(&addrBytes, &ipAddrVal, ipaddress::kIPAddressBytes);
+      // Reverse the bytes to get the correct order. Similar to
+      // IPAddressType. We assume we're ALWAYS on a little endian machine.
+      // Note: for big endian, we should not reverse the bytes.
+      std::reverse(addrBytes.begin(), addrBytes.end());
+      // // Construct a V6 address from the ByteArray16.
+      folly::IPAddressV6 v6Addr(addrBytes);
+
+      // Inline func to get string for ipv4 or ipv6 string
+      auto ipString = [&]() {
+        if (v6Addr.isIPv4Mapped()) {
+          auto v4Addr = v6Addr.createIPv4();
+          return v4Addr.str();
+        }
+        return v6Addr.str();
+      };
+
+      // Format of string is {ipString}/{mask}
+      auto stringRet = fmt::format("{}/{}", ipString(), prefixVal);
+
+      // Write the string to the result vector
+      exec::StringWriter<false> result(flatResult, row);
+      result.append(stringRet);
+      result.finalize();
+    });
+  }
+
+  static void castFromString(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* rowVectorResult = result.as<RowVector>();
+    const auto* ipPrefixStrings = input.as<SimpleVector<StringView>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      auto ipAddressStringView = ipPrefixStrings->valueAt(row);
+      auto ipAddressString = ipAddressStringView.str();
+      auto tryIpPrefix = ipaddress::tryParseIpPrefixString(ipAddressString);
```

Comment:
Are we creating `std::string` here?
From what I understand `ipaddress::tryParseIpPrefixString` uses `folly:StringPiece`, so should we create `folly:StringPiece` instead?